### PR TITLE
Complete t05 /api/article/id .

### PR DESCRIPTION
**Pull request by lanc2112b**
Update articleController.
Update articleModel.
Update app.test.js.
All tests pass.

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -37,4 +37,65 @@ describe("GET Endpoints", () => {
...
+        votes: 0,
+        article_img_url: 'https://images.pexels.com/photos/158651/news-newsletter-newspaper-information-158651.jpeg?w=700&h=700'
+      }];
+
+      return request(app)
+        .get("/api/articles/2")
+        .expect(200)
+        .then((response) => {
+          const { article } = response.body;
+          expect(article).toBeInstanceOf(Array);
```
Comment (yalicee) [inline __tests__/app.test.js]: Should return a single article object

File: __tests__/app.test.js [hunk 2]
```diff
@@ -37,4 +37,65 @@ describe("GET Endpoints", () => {
...
+        article_id: 2,
+        body: 'Call me Mitchell. Some years ago—never mind how long precisely—having little or no money in my purse, and nothing particular to interest me on shore, I thought I would buy a laptop about a little and see the codey part of the world. It is a way I have of driving off the spleen and regulating the circulation. Whenever I find myself growing grim about the mouth; whenever it is a damp, drizzly November in my soul; whenever I find myself involuntarily pausing before coffin warehouses, and bringing up the rear of every funeral I meet; and especially whenever my hypos get such an upper hand of me, that it requires a strong moral principle to prevent me from deliberately stepping into the street, and methodically knocking people’s hats off—then, I account it high time to get to coding as soon as I can. This is my substitute for pistol and ball. With a philosophical flourish Cato throws himself upon his sword; I quietly take to the laptop. There is nothing surprising in this. If they but knew it, almost all men in their degree, some time or other, cherish very nearly the same feelings towards the the Vaio with me.',
```
Comment (yalicee) [inline __tests__/app.test.js]: Please test more dynamically, don't use hardcoded values

File: models/articleModel.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+
+
+
+exports.selectArticleById = (id) => {
+
+  return db.query(`SELECT author, title, article_id, body, topic, created_at, votes, article_img_url
+                  FROM articles
+                  WHERE article_id = $1`, [id])
+    .then((result) => result.rows);
```
Comment (yalicee) [inline models/articleModel.js]: this result.rows is an array. we want to send back an object

**Review by yalicee — CHANGES_REQUESTED** [review]
Please test for 400 and 404 as well.

File: models/articleModel.js (hunk 3 above)
Comment (lanc2112b) [reply models/articleModel.js:9]: And I am sending back an object in the controller lines 28+: 
.then((result) => {
      response.status(200).send({ article: result });
    });
    
on a specified key `article` ?

File: __tests__/app.test.js [hunk 4]
```diff
@@ -38,6 +38,28 @@ describe("GET Endpoints", () => {
...
+      return request(app)
+        .get("/api/articles/2")
+        .expect(200)
+        .then((response) => {
+          const { article } = response.body;
+          expect(article).toBeInstanceOf(Object);
+          expect(article).toMatchObject({
+            author: expect.any(String),
+            title: expect.any(String),
+            article_id: expect.any(Number),
```
Comment (emilyb93) [inline __tests__/app.test.js:53]: I would expect that if i requested article 2, then the article_id should probably be exactly 2

**Review by emilyb93 — APPROVED** [review]
Just change your test to look for the exact article id you request but otherwise this looks good.

**Pull request merged**